stdlib/SparseArrays/src/linalg.jl

```diff
@@ -7,6 +7,18 @@ using Random: rand!
 const StridedOrTriangularMatrix{T} = Union{StridedMatrix{T}, LowerTriangular{T}, UnitLowerTriangular{T}, UpperTriangular{T}, UnitUpperTriangular{T}}
 const AdjOrTransStridedOrTriangularMatrix{T} = Union{StridedOrTriangularMatrix{T},Adjoint{<:Any,<:StridedOrTriangularMatrix{T}},Transpose{<:Any,<:StridedOrTriangularMatrix{T}}}
 
++(A::Hermitian, B::SparseMatrixCSC) = parent(A) + B
++(A::SparseMatrixCSC, B::Hermitian) = A + parent(B)
```

Comment:
Mhm, you're right. What would be the right way to approach that?

Reply:
How about
```julia
function +(A::SparseMatrixCSC, B::Hermitian)
    C = typeof(B.data)(B)
    C .+= A
    return C
end
```
? At least, that preserves potential sparsity of `B.data`.

Reply:
This
```julia
function +(A::SparseMatrixCSC, B::Hermitian{<:Any,<:AbstractSparseMatrix})
    C = sparse(B)
    return A + C
end
function +(A::SparseMatrixCSC, B::Hermitian)
    C = Matrix(B)
    C .= A .+ C
    return C
end
```
seems to solve all cases that I could come up with, including things like `B.data isa UpperTriangular` etc.

Reply:
I apologize for the late replies! Why is that preferable to your earlier one
```julia
function +(A::SparseMatrixCSC, B::Hermitian)
    C = typeof(B.data)(B)
    C .+= A
    return C
end
```
? I'm not sure I see an advantage.

Reply:
Ah, I see it now. `typeof(B.data)(B)` might not necessarily copy.

Reply:
Yeah, you were faster than I could type. Also, it could create some structured matrix type, if that's the type of `B.data`, like an `UpperTriangular`, or a `Bidiagonal` etc. So, basically I think we want either a sparse matrix to dispatch to sparse addition, or a dense matrix that we can add "into". This adds a bit of code (especially for the two orientations of each combination), but I think it's a worthy specialization.

Reply:
I ended up going with `collect` instead of `Matrix`, because I think `Matrix` might not always copy. Does that sound kosher to you?